### ISSUE TITLE
SB2 import: some imports seem broken

### ISSUE THREAD

**Issue by tmickel** [LLK/scratch-vm#501]
### Expected Behavior
Demo:
https://scratch.mit.edu/projects/150198954/
https://llk.github.io/scratch-gui/#150198954

In this project, the math operators looked like this in 2.0:
<img width="186" alt="screen shot 2017-03-14 at 12 21 05 pm" src="https://cloud.githubusercontent.com/assets/120403/23910602/ad94e130-08b0-11e7-9995-bd7df845bc1f.png">
but look like this when imported into 3.0:
<img width="296" alt="screen shot 2017-03-14 at 12 21 32 pm" src="https://cloud.githubusercontent.com/assets/120403/23910625/baf241a6-08b0-11e7-8f24-5052cf9e3ec6.png">
When executed, it appears that they are not running `abs`, either, though (weird).

I believe the problem might be that https://github.com/LLK/scratch-vm/blob/develop/src/import/sb2specmap.js needs updating from https://github.com/LLK/scratch-blocks/pull/829/files; while some blocks were converted from having shadow-inputs to dropdown fields (undroppable), the spec map is still expecting shadow-inputs (e.g., look at `operator_mathop` (previously a shadow-input) vs. `whenKeyPressed` (field)). Not a great failure mode, possibly this case could be detected and cause a throw or something. :)

### Actual Behavior

Selections should match 2.0.

### Steps to Reproduce

Create a project in 2.0 using the math op block and select a math operation. It won't be selected correctly in 3.0

### Operating System and Browser

Mac OS 10.12.1 Chrome latest

**Comment by tmickel** [comment]
(p.s. looks like the VM is actually still running `asin(sin(9))` even though it shows up in the blocks as `abs(abs(9))`. It appears to not be `= 9` because of a rounding/float problem in 3.0, where it `= 8.9999...`).

**Comment by mrjacobbloom** [comment]
Likely an unintended side-effect of LLK/scratch-blocks#829 so you'll probably find the same issue for effects blocks and rotation style

**Comment by thisandagain** [comment]
Thanks @tmickel . Good catch. We'll take a look next week after we get through testing on Thursday (our testing this round doesn't include imports from 2.0).

**Comment by Laphatize** [comment]
I noticed the same issue, it mainly doesn't work with projects that use "custom blocks".

**Comment by tmickel** [comment]
Not sure about custom blocks - those are pretty untouched as far as 3.0 development.

Another thing I noticed though: variables are sometimes broken on the import. That's because they were previously using an input, like this:
```
argMap: [
            {
                type: 'input',
                inputOp: 'data_variablemenu',
                inputName: 'VARIABLE'
            }
        ]
```

But, scratch-blocks was updated to have variables on a particular field:
https://github.com/LLK/scratch-blocks/blob/develop/blocks_vertical/data.js#L66

The mismatch causes weird things, 
<img width="382" alt="screen shot 2017-03-20 at 7 08 44 pm" src="https://cloud.githubusercontent.com/assets/120403/24125779/cb848d0e-0da0-11e7-8e57-818fc428bf63.png">
like this variable called `count` in Scratch 2.0 which appears in the blocks editor as `i`.
